app/src/main/java/io/github/jan/kex/ui/components/ErrorOutLinedTextField.kt

```diff
@@ -0,0 +1,77 @@
+package io.github.jan.kex.ui.components
+
+import androidx.compose.animation.animateColor
+import androidx.compose.animation.core.animateDp
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.core.updateTransition
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import io.github.jan.kex.ui.theme.customColorScheme
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.DurationUnit
+
+@Composable
+fun ErrorOutlinedTextField(
+    modifier: Modifier = Modifier,
+    value: String,
+    onValueChange: (String) -> Unit = {},
+    enabled: Boolean = true,
+    label: @Composable (() -> Unit)?,
+    leadingIcon: @Composable (() -> Unit)?,
+    singleLine: Boolean = false,
+    defaultWidth: Dp = OutlinedTextFieldDefaults.MinWidth,
+    errorExpandWidth: Dp = 30.dp,
+    defaultColor: Color = MaterialTheme.colorScheme.outline,
+    errorColor: Color = customColorScheme.error,
+    errorDisplayTime: Duration = 200.milliseconds,
+    errorDisplayDelay: Duration = 0.milliseconds,
+    displayError: Boolean = false
+) {
+    var isAnimated by remember { mutableStateOf(false) }
+    val transition = updateTransition(targetState = isAnimated, label = "transition")
+
+    val widthOverride by transition.animateDp(transitionSpec = {
+        tween(errorDisplayTime.toInt(DurationUnit.MILLISECONDS), errorDisplayDelay.toInt(DurationUnit.MILLISECONDS))
+    }, "") { animated ->
+        if (animated) errorExpandWidth + defaultWidth else defaultWidth
+    }
+    val colorOverride by transition.animateColor(transitionSpec = {
+        tween(errorDisplayTime.toInt(DurationUnit.MILLISECONDS) /*errorDisplayDelay*/)
+    }, "") { animated ->
+        if (animated) errorColor else defaultColor
+    }
+
+    isAnimated = displayError
```

Comment:
```suggestion
    var isAnimated by remember(displayError) { mutableStateOf(displayError) }
    val transition = updateTransition(targetState = isAnimated, label = "transition")

    val widthOverride by transition.animateDp(transitionSpec = {
        tween(errorDisplayTime.toInt(DurationUnit.MILLISECONDS), errorDisplayDelay.toInt(DurationUnit.MILLISECONDS))
    }, "") { animated ->
        if (animated) errorExpandWidth + defaultWidth else defaultWidth
    }
    val colorOverride by transition.animateColor(transitionSpec = {
        tween(errorDisplayTime.toInt(DurationUnit.MILLISECONDS) /*errorDisplayDelay*/)
    }, "") { animated ->
        if (animated) errorColor else defaultColor
    }
```
Man könnts wahrscheinlich so machen, dann ist der reassign nicht mehr so komisch da

Reply:
Achso und was sind diese leeren strings bei den transitions?

Reply:
bzw warum gibts überhaupt die isAnimated variable, die ist doch im grunde einfach die displayError variable?

Reply:
ka ich hab einfach das example copiert

Reply:
ja geht auch ohne